### PR TITLE
[Go] conditionals - replace about.md file with concept files

### PR DESCRIPTION
**Pull request by ErikSchierboom**
In [this issue](https://github.com/exercism/v3/issues/2293) we're describing an evolution of the specification of this repo.
These changes include, amongst others, replacing the contents of the `after.md` document with individual concept documents. There are two documents per concept:

1. An `about.md` file containing the description of the concept
1. A `links.json` file containing a set of useful links related to the concept. 

This looks as follows in the repo:

```
<track>
├── concepts
│   ├── <concept>
│   │   ├── about.md
│   │   └── links.json
│   └── ...
```

This PR applies this change to this exercise by: 
                    
1. Copying the existing `after.md` file's contents to the `about.md` file of each concept listed in the exercise's `concepts` section in the `config.json` file.
1. Extract the links from the `after.md` file and put them into the `links.json` file

Before merging this PR, please check that the contents of the concept documents makes sense.
This is especially true when the exercise unlocks multiple concepts, as then the concept documents should be updated to only contain the information relevant to that concept.

Note: to make reviewing easier, the PR has been split into three separate commits. Please squash this PR upon merging.

### CODE REVIEW

**Review by oanaOM — APPROVED** [review]
@ErikSchierboom , The only thing I've noticed here is that multiple files for different exercises were committed here. Other than this, the files looks ok so I assume we can merge them, right?

**Comment by ErikSchierboom** [comment]
@oanaOM I'm cross-linking my comment on the other Go PR: https://github.com/exercism/v3/pull/2357#issuecomment-717251226

TL;DR As the `conditionals` exercise unlocks multiple concepts, there are multiple concept documents. It is the task of the Go maintainers to edit the concept documents to contain just the information of that concept. So I fully expect the Go maintainers to update the files in this PR :)

**Review by tehsphinx — APPROVED** [review]
@ErikSchierboom: How's this?

**Comment by ErikSchierboom** [comment]
@tehsphinx Looks good! I think the only change I would suggest is to remove the headings from the `about.md` files. They are redundant. There is also a merge conflict.

**Comment by tehsphinx** [comment]
@ErikSchierboom: What now?

**Comment by ErikSchierboom** [comment]
Thanks @tehsphinx!
